Docs/Deployent and Data Tier Sizing (#882)
grouparoo/www.grouparoo.com: pages/docs/deployment/monitoring-and-performance.mdx

Subsequent Runs will be much faster. Grouparoo can cache the [Record Properties](/docs/getting-started/product-concepts#record-property), caches [high watermarks from Schedule Runs](/docs/getting-started/product-concepts#high-watermark), and if nothing has changed, Grouparoo will not need to re-import or re-export those Records until they do change.

```
1,000,000 Records * (0.1s + 0.1s)
--------------------------------- = ~11 hours
60 s/min * 60 min/hr * 5 workers
```

If your Grouparoo sync is going slower than the above example, here are suggestions to speed things up. These are listed in order of likelihood to help.

1. Ensure that your Sources are adequately provisioned for repeated large reads. Check your logs, query times, and vendor credits.
2. [Ensure the Grouparoo DB is fast enough](/docs/deployment/network#postgres)
3. [Slowly increase the Grouparoo Batch Sizes](/docs/deployment/environment#batch-size-environment-variables)
4. Check if Grouparoo is hitting rate limits when exporting Records to your Destinations. Is there a larger plan available with more throughput?

> 5. Slowly increase the number if [`WORKERS`](/docs/deployment/environment#required-environment-variables) "threads" the Grouparoo application uses. The number of workers can be thought of as internal processing "threads", each working in parallel to import and export your data. Increasing the number of `WORKERS` may speed up Grouparoo, provided your sources and destinations can handle the increased volume and load. If not, increasing the number of `WORKERS` will actually slow things down. An upper limit of 20 WORKERS per server should be maintained.

Reviewer: ```suggestion
5. Gradually increase the number of [`WORKERS`](/docs/deployment/environment#required-environment-variables) "threads" the Grouparoo application uses. The number of workers can be thought of as internal processing "threads", each working in parallel to import and export your data. Increasing the number of `WORKERS` may speed up Grouparoo, provided your sources and destinations can handle the increased volume and load. If not, increasing the number of `WORKERS` might actually slow things down. An upper limit of 20 WORKERS per server should be maintained.
```
